transitions/core.py

```diff
@@ -359,16 +356,17 @@ def __init__(self, model=None, states=None, initial=None, transitions=None,
         self._transition_queue = deque()
         self.models = []
 
-        if initial is None:
-            self.add_states('initial')
+        if model is None and add_self:  # TODO(pbovbel) reconsider API for next major release
+            model = self
+
```

Comment:
While we definitely want this possible I don't think it is the most typical use case. What about just reversing the logic of `add_self` so that there is a `dont_add_self` (with a much better name than that) that would prevent the `Machine` being added as a model but the default is to add `self` so the API doesn't break and people can use a `Machine` in a straight-forward way.